### PR TITLE
feat: add function: LedgerAlignBuffer

### LINKED ISSUE #113: Aligning all lines
`LedgerAlign()` currently takes an optional range of lines, defaulting to the current line. From a usability perspective, I think it would be useful to default to align all lines unless a range is given.

### PR DESCRIPTION
**Pull request by sudoforge**
closes #113

### CODE REVIEW

**Comment by sudoforge** [comment]
Initially, I changed `LedgerAlign` to default to a range of `%`, or "all lines". In thinking about the impact of this, it's possible that running the function in this way causes an undesirable performance hit on large files.

I break my entries up by year and month, and none of my files ever get very large. I'd love to have users who have large files run this and let me know if there are performance impacts.

File: ftplugin/ledger.vim [hunk 1]
```diff
@@ -508,6 +508,8 @@ command! -buffer -nargs=+ -complete=customlist,s:autocomplete_account_or_payee
 
 command! -buffer -range LedgerAlign <line1>,<line2>call ledger#align_commodity()
 
+command! -buffer LedgerAlignAll call ledger#align_all_commodity()
```
Comment (alerque) [inline ftplugin/ledger.vim:511]:
```suggestion
command! -buffer LedgerAlignBuffer call ledger#align_commodity_buffer()
```

Is there a precedent for `...All` functions that act on the entire buffer or would this be more explicit? In any case the word order should be the same for the internal command.

**Review by alerque — CHANGES_REQUESTED** [review]
I don't mind a utility function like this to simplify the steps to operate on a whole buffer. This grinds to a halt (taking *minutes* to complete) on my ledgers, but they are not small and I don't think having the function around for those that want a binding for it well affect existing workflow(s).

1. Review function naming.
2. Document in the `:help` docs.

File: ftplugin/ledger.vim (hunk 1 above)
Comment (sudoforge) [reply ftplugin/ledger.vim]: Resolved in 0fc0d2e.

Comment (sudoforge) [reply ftplugin/ledger.vim]: > Is there a precedent for ...All functions that act on the entire buffer or would this be more explicit? 

Not that I'm particularly aware of, although I should also note that I haven't really delved into contributing to rules in the past so my experience is limited. I liked your suggestion of `LedgerAlignBuffer`, though.

**Review by alerque — APPROVED** [review]

File: doc/ledger.txt [hunk 2]
```diff
@@ -135,6 +135,13 @@ Tips and useful commands
   default commodity is set to `'$'`). Or you may press <Tab> in Visual mode
   to align a number of transactions at once.
 
+* `:LedgerAlignBuffer`
```
Comment (alerque) [inline doc/ledger.txt:138]: Lets move this up in the help to be just below it's `:LedgerAlign` counterpart.

Also lets mention very briefly in the help blurb *why* this is different than just `:%call ledger#align_commodity()`, namely that it saves and restores the cursor position.

**Review by alerque — CHANGES_REQUESTED** [review]
Sorry to bother you with another edit, but I think the help blurb would serve better with two minor tweaks...

File: doc/ledger.txt (hunk 2 above)
Comment (sudoforge) [reply doc/ledger.txt]: Resolved in facc97f.

**Review by alerque — APPROVED** [review]

**Comment by alerque** [comment]
Thanks for taking the time to contribute and working through change requests. This looks good to me now.

**Pull request merged**